### PR TITLE
Refactor code logic of functionality about getting rule doc location

### PR DESCRIPTION
**Pull request by tren-parasoft**

### CODE REVIEW

**Comment by tren-parasoft** [comment]
This pull request is ready for review now.

**Review by rwang-parasoft — CHANGES_REQUESTED** [review]
Handle the situation when dtp service is not avaliable.

**Comment by rwang-parasoft** [comment]
I am verifying.
The result is fine.

**Review by rwang-parasoft — APPROVED** [review]

File: parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcess.java [hunk 1]
```diff
@@ -161,7 +163,7 @@ private void doWork() {
         String settingsPath = params.get(SETTINGS_LOCATION);
         Properties properties = _localSettingsHelper.loadLocalSettings(checkoutDir, settingsPath);
         if (!properties.isEmpty()) {
-            if (_localSettingsHelper.isDtpUrlValidForTeamCity(properties.getProperty("dtp.url"))) {
+            if (_localSettingsHelper.isDtpUrlValidForTeamCity(properties.getProperty(DTP_URL))) {
```
Comment (yshi-parasoft) [inline parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcess.java]: Can you change `DTP_URL` to `LocalSettingsHelper.DTP_URL` and remove the import?

File: parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/RuleDocumentationUrlProvider.java [hunk 2]
```diff
@@ -16,30 +16,36 @@
...
-    private final TeamCityRuleDocumentationProvider _docProvider;
+    private final RuleDocumentationProvider _docProvider;
 
-    public RuleDocumentationUrlProvider(AgentRunningBuild build, Properties settings) {
+    public RuleDocumentationUrlProvider(AgentRunningBuild build, Properties settings ) {
         _build = build;
         _ruleDocsUrls = new HashMap<>();
-        _docProvider = new TeamCityRuleDocumentationProvider(settings);
+        _docProvider = new RuleDocumentationProvider(settings);
+        if (_docProvider.getDtpDocServiceStatus() == ClientStatus.NOT_AVAILABLE) {
```
Comment (yshi-parasoft) [inline parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/RuleDocumentationUrlProvider.java:36]: Is it better to add a string field for description of error? In that way we don't need to write the message in different projects.
```
public enum ClientStatus {
    AVAILABLE("DTP server is available"),
    NOT_SUPPORTED_VERSION("Unable to retrieve the documentation for the rule from DTP. Please ensure that a DTP version older than 2023.1 is used"),
    NOT_AVAILABLE("DTP server is not available"),
    DTP_URL_NOT_SPECIFIED("DTP URL is not specified");

    private final String description;

    ClientStatus(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return description;
    }
}

```

**Pull request merged**